### PR TITLE
Convert to Polymer 3, add events type definitions

### LINKED ISSUE #548: Polymer 3 conversion and TS event type definitions
### Description

- convert codebase to Polymer 3
- use `registerStyles` API instead of `<dom-module theme-for>`
- add events type definitions

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #548

### CODE REVIEW

File: theme/material/vaadin-text-field-styles.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      );
+      background-size: 4px 1px;
+      background-repeat: repeat-x;
+    }
+
+    /* Only target the visible floating label */
+    :host([has-label]:not([has-value]):not([focused]):not([invalid]):not([theme~='always-float-label']))
+      [part='label'] {
+      width: 100%;
+      /* IE11 doesn’t work with calc inside the translate function, so we need to have a fixed pixel value instead of 50% + 16px */
```
Comment (alvarezguille) [inline theme/material/vaadin-text-field-styles.js:190]: Remove IE11 workaround

File: src/vaadin-text-field-mixin.js [hunk 2]
```diff
@@ -441,11 +435,7 @@
     // of the clear button instead of setting display property for it depending on state.
     /** @private */
     _stateChanged(disabled, readonly, clearButtonVisible, hasValue) {
-      if (!disabled &&
-          !readonly &&
-          clearButtonVisible &&
-          hasValue
-      ) {
+      if (!disabled && !readonly && clearButtonVisible && hasValue) {
```
Comment (alvarezguille) [inline src/vaadin-text-field-mixin.js:438]: Remove IE11 workaround or adjust comment

**Review by alvarezguille — CHANGES_REQUESTED** [review]
Some comments about IE11 workarounds, LGTM otherwise

File: theme/material/vaadin-text-field-styles.js (hunk 1 above)
Comment (web-padawan) [reply theme/material/vaadin-text-field-styles.js]: I tried to use 50% + 16px and that caused regressions in visual tests. Removed the comment for now.

Let's revisit this later, as Material theme in general is not up to date.

File: src/vaadin-text-field-mixin.js (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-text-field-mixin.js]: Removed the workaround.

**Review by alvarezguille — APPROVED** [review]

**Review by alvarezguille — APPROVED** [review]

**Pull request merged**